api/src/services/extensions.ts

```diff
@@ -245,18 +245,19 @@ export class ExtensionsService {
 	 *    - Entry status change resulted in all children being disabled then the parent bundle is disabled
 	 *    - Entry status change resulted in at least one child being enabled then the parent bundle is enabled
 	 */
-	private async checkBundleAndSyncStatus(trx: Knex, bundleId: string, extension: ApiOutput) {
+	private async checkBundleAndSyncStatus(trx: Knex, extensionId: string, extension: ApiOutput) {
 		if (extension.bundle === null && extension.schema?.type === 'bundle') {
 			// If extension is the parent bundle, set it and all nested extensions to enabled
 			await trx('directus_extensions')
 				.update({ enabled: extension.meta.enabled })
-				.where({ bundle: bundleId })
-				.orWhere({ id: bundleId });
+				.where({ bundle: extensionId })
+				.orWhere({ id: extensionId });
 
 			return;
 		}
 
-		const parent = await this.readOne(bundleId);
+		const parentId = extension.bundle ?? extensionId;
```

Comment:
What is this fallback to `extensionId` for?

Reply:
Ah yes, it should be `extension.meta.bundle`. 🙈 
Added the fallback just in case. 

Further processing will be skipped if `parentId` is `null`. 
It's patched via https://github.com/directus/directus/pull/22587/commits/5fd3a5b19cd9ed71db0fb3321aa134122a9ce5b2.

Reply:
Perfect 🚀❤️